tools/dev-doctor/rootcmd.go

```diff
@@ -119,6 +123,34 @@ func rootCmdRun(cmd *cobra.Command, args []string) {
 		dockerGroupCheck{},
 	}
 
+	if *backportingChecks {
+		checks = append(checks,
+			&binaryCheck{
+				name:       "jq",
+				ifNotFound: checkError,
+			},
+			&binaryCheck{
+				name:          "python3",
+				ifNotFound:    checkError,
+				versionArgs:   []string{"--version"},
+				versionRegexp: regexp.MustCompile(`Python\s+(\d+.\d+\.\d+)`),
+				minVersion:    &semver.Version{Major: 3, Minor: 6, Patch: 0},
+			},
+			&commandCheck{
+				name:             "pygithub",
+				command:          "python3",
+				args:             []string{"-c", "from github import Github"},
+				ifFailure:        checkWarning,
+				ifSuccessMessage: "pygithub installed",
+				hint:             `Run "pip3 install --user PyGithub".`,
+			},
+			&envVarCheck{
+				name:            "GITHUB_TOKEN",
+				ifNotSetOrEmpty: checkWarning,
+			},
```

Comment:
Not sure we want to include that check. I usually pass that environment variable inline (i.e., `GITHUB_TOKEN=xxx ./submit-backport`) and that's what our backporting guide currently recommends.

Plus, if we're not actually checking the token works, I'm unsure what the value of this check is. All commands in the guide already have that token set when necessary, so it should be clear that it's needed.

Reply:
In [my review comment above](https://github.com/cilium/cilium/pull/14016#discussion_r522879450) I didn't consider the case that might people pass `GITHUB_TOKEN` inline. In my workflow, I usually export it once in my shell session before doing backports.

I agree with @pchaigno that we thus shouldn't make this mandatory.

Reply:
I think the check is still useful as you need the environment variable to be set somehow. To balance the two sides I've introduced a new check result `info` which prints information but does not affect the overall success/failure of the check. In this case you'll get:

```
RESULT    CHECK            MESSAGE
info      github-token     $GITHUB_TOKEN not set or empty
```

and the overall `dev-doctor` command still exits with success.

Does this seem like a reasonable compromise?

Reply:
Having this check as `info` SGTM, thanks.

Reply:
LGTM to me too.